File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/loader/ILoader.java
package ru.kmorozov.library.data.loader;

import ru.kmorozov.library.data.model.book.Book;

import java.io.IOException;

/**
 * Created by sbt-morozov-kv on 14.03.2017.
 */
public interface ILoader {

    void load() throws IOException;

    void resolveLink(Book lnkBook);

    String downloadBook(Book book);
}

File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/loader/ILoader.kt
package ru.kmorozov.library.data.loader

import ru.kmorozov.library.data.model.book.Book

import java.io.IOException

/**
 * Created by sbt-morozov-kv on 14.03.2017.
 */
interface ILoader {

    @Throws(IOException::class)
    fun load()

    fun resolveLink(lnkBook: Book)

    fun downloadBook(book: Book)
}
